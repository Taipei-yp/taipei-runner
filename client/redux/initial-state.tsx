const initialState = {
  user: {},
  forum: {},
  leaderboard: {},
};
export default initialState;
