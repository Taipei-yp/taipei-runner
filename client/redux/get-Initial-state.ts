import { RouterState } from "connected-react-router";
import { initialState as auth } from "./auth/auth-reducer";
import { initialState as leaderboard } from "./leaderboard/leaderboard-reducer";
import { initialState as profile } from "./profile/profile-reducer";
import { RootState } from "./root-reducer";
import { initialState as theme } from "./theme/theme-reducer";

export const getInitialState = (pathname = "/"): RootState => {
  return {
    auth,
    profile,
    leaderboard,
    router: {
      location: { pathname, search: "", hash: "", key: "" },
      action: "POP",
    } as RouterState,
    theme,
  };
};
