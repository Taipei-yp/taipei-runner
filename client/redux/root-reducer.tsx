import { combineReducers } from "redux";
import { forumReducer } from "./forum/forum-reducer";
import { leaderboardReducer } from "./leaderboard/leaderboard-reducer";
import { userReducer, UserState } from "./user/user-reducer";

export type RootState = {
  user: UserState;
};

export const rootReducer = combineReducers({
  user: userReducer,
  forum: forumReducer,
  leaderboard: leaderboardReducer,
});
