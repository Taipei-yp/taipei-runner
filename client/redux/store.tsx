import { createBrowserHistory, createMemoryHistory } from "history";
import { applyMiddleware, compose, createStore } from "redux";
import thunk from "redux-thunk";
import {
  authApi,
  forumApi,
  leaderboardApi,
  oauthApi,
  profileApi,
  themeApi,
} from "client/api";
import { feedbackApi } from "client/api/feedback-api";
import { isServer } from "client/helpers/check-server";
import { getInitialState } from "./get-Initial-state";
import { createRootReducer, RootState } from "./root-reducer";

export type Api = {
  authApi: typeof authApi;
  profileApi: typeof profileApi;
  oauthApi: typeof oauthApi;
  leaderboardApi: typeof leaderboardApi;
  themeApi: typeof themeApi;
  forumApi: typeof forumApi;
  feedbackApi: typeof feedbackApi;
};

const api: Api = {
  authApi,
  profileApi,
  oauthApi,
  leaderboardApi,
  themeApi,
  forumApi,
  feedbackApi,
};

const configureStore = (
  initialState: RootState = getInitialState(),
  url = "/",
) => {
  const history = isServer
    ? createMemoryHistory({ initialEntries: [url] })
    : createBrowserHistory();

  const rootReducer = createRootReducer(history);

  const store = createStore(
    rootReducer,
    initialState,
    compose(
      applyMiddleware(thunk.withExtraArgument(api)),
      !isServer && (window as any).__REDUX_DEVTOOLS_EXTENSION__
        ? (window as any).__REDUX_DEVTOOLS_EXTENSION__()
        : (f: () => void) => f,
    ),
  );

  return { store, history };
};

export { configureStore };
