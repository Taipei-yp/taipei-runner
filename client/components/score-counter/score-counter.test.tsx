import React from "react";
import { Provider } from "react-redux";
import TestRenderer from "react-test-renderer";
import { configureStore } from "client/redux/store";
import { ScoreCounter } from ".";

describe("Component ScoreCounter", () => {
  const { store } = configureStore();

  const scoreCounter = TestRenderer.create(
    <Provider store={store}>
      <ScoreCounter score={1000} />
    </Provider>,
  );

  it("correct render", () => {
    expect(scoreCounter.toJSON()).toMatchSnapshot();
  });
});
