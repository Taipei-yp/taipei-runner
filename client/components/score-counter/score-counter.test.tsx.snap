// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ScoreCounter correct render 1`] = `
<span
  className="score-counter"
>
  1000
</span>
`;
